gptme/server/tasks_api.py

```diff
@@ -0,0 +1,723 @@
+
+    try:
+        # Generate task ID
+        task_id = f"task-{datetime.now().strftime('%Y%m%d-%H%M%S')}"
+
+        # Create task
+        task = Task(
+            id=task_id,
+            content=req_json["content"],
+            created_at=datetime.now().isoformat(),
+            status="pending",
+            target_type=req_json.get("target_type", "stdout"),
+            target_repo=req_json.get("target_repo"),
+            conversation_ids=[],
+            metadata=req_json.get("metadata", {}),
+        )
+
+        # Create initial conversation
+        conversation_id = create_task_conversation(task)
+        task.conversation_ids = [conversation_id]
+
+        # Save task
+        save_task(task)
+
+        logger.info(f"Created task {task_id} with conversation {conversation_id}")
+
+        return flask.jsonify(get_task_info(task)), 201
+
+    except Exception as e:
+        logger.error(f"Error creating task: {e}")
+        return flask.jsonify({"error": str(e)}), 500
+
+
+@tasks_api.route("/api/v2/tasks/<string:task_id>")
+def api_tasks_get(task_id: str):
+    """Get detailed task information."""
+    task = load_task(task_id)
+    if not task:
+        return flask.jsonify({"error": f"Task not found: {task_id}"}), 404
+
+    try:
+        original_status = task.status
+        task_info = get_task_info(task)
+
+        # Persist updated status if it changed
+        if task.status != original_status:
+            save_task(task)
+            logger.info(
+                f"Updated task {task_id} status from {original_status} to {task.status}"
+            )
+
+        return flask.jsonify(task_info)
+    except Exception as e:
+        logger.error(f"Error getting task {task_id}: {e}")
+        return flask.jsonify({"error": str(e)}), 500
+
+
+@tasks_api.route("/api/v2/tasks/<string:task_id>", methods=["PUT"])
+def api_tasks_update(task_id: str):
+    """Update task metadata."""
+    task = load_task(task_id)
+    if not task:
+        return flask.jsonify({"error": f"Task not found: {task_id}"}), 404
+
+    req_json = flask.request.json
+    if not req_json:
+        return flask.jsonify({"error": "No JSON data provided"}), 400
+
+    try:
+        # Update allowed fields
+        if "content" in req_json:
+            task.content = req_json["content"]
+        if "target_type" in req_json:
+            task.target_type = req_json["target_type"]
+        if "target_repo" in req_json:
+            task.target_repo = req_json["target_repo"]
+        if "metadata" in req_json:
+            task.metadata.update(req_json["metadata"])
+
+        save_task(task)
+
+        return flask.jsonify(get_task_info(task))
+
+    except Exception as e:
+        logger.error(f"Error updating task {task_id}: {e}")
+        return flask.jsonify({"error": str(e)}), 500
+
+
+@tasks_api.route("/api/v2/tasks/<string:task_id>", methods=["DELETE"])
+def api_tasks_delete(task_id: str):
```

Comment:
Not sure we want this, or at least not have it work like this. Maybe add a "archive" functionality instead, which doesn't actually delete anything, just marks it as archived (could be either a pending/active or done task that we no longer care about).